### DISCUSSION TITLE
glb model floating?

### DISCUSSION THREAD

**Discussion by veronicamontezuma** [google/model-viewer#1966] [Q&A]
Hi! I'm having an issue that I'm not quite sure where it might come from...
My glb model is showing up hovering above the floor just enough for it to be weird. I understand model-viewer adds a small offset (for the shadow, I believe?), but I think something is off here, because the usdz file is positioned correctly on the floor (and is converted from the same .glb that floats).
Does anyone know what might be causing this? Or if there's a way to remove the model-viewer offset, perhaps?

**Comment by elalish** [comment]
Can you share a screen shot? Indeed, the USDZ does not add the shadow offset, so that is a difference, but it should be quite small. Have you verified that your model is the right size in meters? That offset will scale up with model height, and it's sometimes unclear that the model is gargantuan.

**Reply by veronicamontezuma** [reply]
Ah, I didn't realize the USDZ had no offset, that clarifies this part! In theory the model is the correct size, we tried to be careful with that because the project involves furniture. But perhaps it's big enough that the offset shows?

Here's a screenshot:
![WhatsApp Image 2021-01-25 at 18 20 07](https://user-images.githubusercontent.com/11477166/105767631-031e6e80-5f3a-11eb-9561-b9f1aa75cd9a.jpeg)

**Reply by veronicamontezuma** [reply]
oh s*** sorry for the MASSIVE photo, didn't think this trough very well

**Comment by elalish** [comment]
No prob. Have you tried `bounds="tight"`? https://modelviewer.dev/docs/index.html#entrydocs-stagingandcameras-attributes-bounds

I just added this feature in v1.4 since I realized the bounding box computation by three.js can be overlarge sometimes.

**Reply by veronicamontezuma** [reply]
I actually did try it, but nothing changed...
I asked the person who made the 3D model about the dimensions, to make sure that is not the issue (but should only hear back tomorrow, damn time zones).

Do you know if there's a way to override the offset, somehow? We thought of editing the glb or even the model directly (lowering it a bit to compensate the offset) but it sounded like the odd way to go about this (since the usdz would need a different positioning and then it all starts to become too much improvisation – but still, "doable"...)

**Reply by elalish** [reply]
Is the model animated? That can sometimes throw off the bounding box computation. Do you have a link you can share? It's possible this is a bug.

**Reply by veronicamontezuma** [reply]
It's not animated, no. 
We have a test page running here: [https://nookdesign.pixelar.studio](https://nookdesign.pixelar.studio)
If you need to see the code better, I can try to set up a glitch or something similar.

**Reply by elalish** [reply]
I can repro; this looks like a bug in `bounds="tight"`. Feel free to file an issue.

**Reply by veronicamontezuma** [reply]
okay, will do. thank you!!

**Reply by elalish** [reply]
Nevermind, this appears to be a SceneViewer issue. `bounds="tight"` does fix it for the 3D mode shadow and also for WebXR AR mode.
